City-of-Helsinki/helsinki-paatokset
# PP-182: Language version handling with Ahjo data

## What was done:
- Fixes to cron commands
- Better logging for for callbacks, should help with debugging
- Better local proxy functionality, should now be possible to migrate (most) data locally through the test site instead of relying on the VPN
- Functionality to display policymakers, trustees and decisions on finnish and swedish language versions even if they are not translated
- Functionality to hide "translated" decisions if a version exists in the currently selected language

## How to test
- Checkout branch
- Read code changes. Anything weird that should be refactored?
- Edit your `.env.local` file and change the `AHJO_PROXY_BASE_URL` line to this: `AHJO_PROXY_BASE_URL=https://nginx-paatokset-test.agw.arodevtest.hel.fi/`
  - Add this line to the end: `SKIP_AUTH_HEADERS=1` 
- Run `make new`
- Login to the container and run:
  - `printenv` to test if the proxy URL variable was updated 
  - `drush mim ahjo_policymakers:all;drush mim ahjo_trustees:council;drush mim ahjo_meetings:latest;drush mim ahjo_cases:latest;drush mim ahjo_decisions:latest`
  - `drush ac:l` to check that the queue is empty
- Check the decisions list: https://helsinki-paatokset.docker.so/fi/admin/content/decisions
  - None of the dates should be set yet, and there shouldn't be any swedish language decisions
- Fetching decision data:
  - First run `drush ap:ud -v --limit=1` (note that we're not fetching local data)
  - If the run should not succeed (missing cases and meetings is fine), run it again just in case. Contact Juho if you can't get this to work
  - If the previous command succeeds, run `drush ap:ud -v` to update all decision data
  - Reload the decisions list. Now the previously missing values should be there, and there should be some swedish language decisions
  - There should be some items in the callback list now: `drush ac:l`
  - Process the list with `drush queue:run ahjo_api_subscriber_queue`
- Run these commands to get good test data for language versioning:
```
drush ap:update meetings 0290020221 -v
drush ap:update cases HEL-2022-000553 -v
drush ap:update cases HEL-2021-013876 -v
drush ap:update cases HEL-2021-013311 -v
drush ap:update cases HEL-2021-013090 -v
drush ap:update cases HEL-2021-013729 -v
drush ap:update cases HEL-2016-005582 -v
drush ap:update cases HEL-2021-000672 -v
drush ap:update cases HEL-2021-012096 -v
drush ap:update cases HEL-2021-012759 -v
drush ap:update cases HEL-2020-001640 -v
drush ap:update cases HEL-2017-003866 -v
drush ap:update cases HEL-2013-013706 -v
drush ap:update cases HEL-2021-005589 -v
drush ap:update cases HEL-2021-006774 -v
drush ap:update decisions {777921F2-B39B-4F26-9434-2FAE4CE45DBC} -v
drush ap:update decisions {2146ABFC-EA7E-4CA4-9026-800B479D0EDA} -v
drush ap:update decisions {3AE8BD99-C92A-4568-9BAF-744CE791BC0E} -v
drush ap:update decisions {766D79C7-C25D-43A3-96E4-B5504DB50802} -v
drush ap:update decisions {06DC4894-4979-47C6-9CCD-4CDDFF13BD00} -v
drush ap:update decisions {9EE6C451-9922-437D-8F87-F891D7DF5C7B} -v
drush ap:update decisions {C780C248-2C80-4213-B9A5-01F591C9EDD6} -v
drush ap:update decisions {6E532934-8197-452A-8B37-D3855AAAA700} -v
drush ap:update decisions {EC075C20-DD8C-42F0-8800-48EFEF322187} -v
drush ap:update decisions {340FAF4B-094F-425F-8203-E05EDB22E635} -v
drush ap:update decisions {84B9C816-75C9-4C27-9B41-2D693BC652F5} -v
drush ap:update decisions {1DB88F4C-2C50-4453-8608-B620651AF5D1} -v
drush ap:update decisions {F064A1AD-848E-4042-9D08-1CB811D7A838} -v
drush ap:update decisions {E7F52B0C-EBA5-41E8-B60F-E21747BAB19A} -v
drush ap:update decisions {7D420E39-7782-4E81-A35E-40C0C5171852} -v
drush ap:update decisions {71F95126-8721-4252-991B-BD69C5CC3980} -v
drush ap:update decisions {CB63EA0E-D23B-4565-8EAC-4A1ED84C7CAB} -v
drush ap:update decisions {AB7E74D4-CA30-47D0-8146-9E20B20CB292} -v
drush ap:update decisions {16E7C9D8-07FF-421F-845E-D6E96035DA18} -v
drush ap:update decisions {6DBBC1A6-EF61-42DB-AB2F-F636B3726D4E} -v
drush ap:update decisions {48296C43-F032-4469-A9D9-DE1BD472FDE1} -v
drush ap:update decisions {446B17F9-7B50-4D9A-A342-F4AD530E480A} -v
drush ap:update decisions {610BB693-74E5-4165-B3EF-9AFBA1B23EC9} -v
drush ap:update decisions {54B7840A-768D-4505-B6DE-D28C6C8971D9} -v
drush ap:update decisions {6EEEBB31-A08B-43A7-BD84-1DD567C4DE99} -v
drush ap:update decisions {8C041C0E-70DD-43CB-9F89-FD4954346435} -v
drush ap:update decisions {54858DFE-36E8-4F5E-9BDC-B94812725612} -v
drush ap:update decisions {0454D131-B45C-4967-9106-C9DD153524C3} -v
drush ap:update decisions {362F21D9-C1C7-431D-87F8-5A5E96BFECEC} -v
drush ap:update decisions {3A02B66F-11F8-4DA3-A26D-DA245E033AA6} -v
drush ap:update decisions {63506843-41D1-41C3-B24B-AF7583DB08C9} -v
drush ap:update decisions {7F8DB767-E45C-4307-9176-FD16D3ADBE9F} -v
drush ap:update decisions {0A4E2E32-0545-4FD3-97D2-D6702DB4267F} -v
drush ap:update decisions {08012F40-4B03-44C3-BB73-D9B923F4FF50} -v
```

### Policymaker functionality
- Compare the finnish and swedish Kaupunginvaltuusto pages:
  - https://helsinki-paatokset.docker.so/fi/paattajat/kaupunginvaltuusto and https://helsinki-paatokset.docker.so/sv/beslutsfattare/kaupunginvaltuusto
  - Test that the meeting documents lists and meeting minutes pages work
  - Also test that the side navigation works on both languages
  - Translating a policymaker page should work and the translated fields should show up normally (API fields can't be translated)
- Same should apply for an official's page:
  - https://helsinki-paatokset.docker.so/fi/paattajat/kaupunkiympariston-toimialajohtaja-kaupunkiympariston-toimiala and https://helsinki-paatokset.docker.so/sv/beslutsfattare/kaupunkiympariston-toimialajohtaja-kaupunkiympariston-toimiala  
- Check this Kaupunginvaltuusto meeting minutes page:
  - https://helsinki-paatokset.docker.so/fi/paattajat/kaupunginvaltuusto/asiakirjat/0290020221 and https://helsinki-paatokset.docker.so/sv/beslutsfattare/kaupunginvaltuusto/dokumenter/0290020221
  - Only current language decisions should be visible on the selected language
  - Clicking on a decision link should take you to the decisions with the same title on the correct language version

### Decisions functionality
- Check that you get a different default decision for both of these links:
  - https://helsinki-paatokset.docker.so/fi/asia/hel-2021-013311 and https://helsinki-paatokset.docker.so/sv/arende/hel-2021-013311
- Test the decisions in the previous kaupunginvaltuusto meeting
  - The swedish versions should not appear in the dropdown or next / previous buttons
  - The dropdown should be working as intended
- Test some other decisions. You can fetch more data with `drush mim ahjo_decisions:all;drush ap:ud -v`
- We can do more in depth testing on the test site because it has more content.

### Trustee functionality
- Check a trustee page on both languages:
  - https://helsinki-paatokset.docker.so/fi/paattajat/arhinpa and https://helsinki-paatokset.docker.so/sv/beslutsfattare/arhinpa
  - Add an image and enter some fields to the trustee, make sure they appear on both language versions
  - Try translating the trustee page to swedish. All the custom fields should be able to be translated normally (except for the API content)
